src/plugins/NEMD/RegionSampling.h

```diff
@@ -136,7 +136,7 @@ class SampleRegion : public CuboidRegionObs
 	// instances / ID
 	static unsigned short _nStaticID;
 
-	uint32_t _numComponents;
+	uint8_t _numComponents;
```

Comment:
I'm still confused as to what the type of numComponents should be. Here it is `uint8_t`, in MettDeamon.cpp:1275 it is a `uint16_t` (created from a `size_t` which is basically a `uint64_t` and used as such) and later in MettDeamon.cpp:1520 it is an `unsigned int` which is typically a `uint32_t`. Does the type actually matter? If not I would suggest just always using `size_t` for simplicity / readability.

Reply:
Yes, type consistivity is a big problem... As far as I know, the components are stored in a vector, so something like components.size() returns a data of type size_t. Means, I would support your suggestion.

Reply:
> To be honest, it would be a lot of work to change all `_numComponents`, `cid`, etc. to `size_t` or `auto`

True, this goes beyond this PR but I think it's good to keep in mind for future development and not make random changes. Maybe at some point, someone who is responsible for "software sustainability" should clean this up ;)